src/coreclr/tools/Common/TypeSystem/Interop/UnmanagedCallingConventions.cs

```diff
@@ -135,6 +135,25 @@ public static UnmanagedCallingConventions GetPInvokeMethodCallingConventions(thi
             return result;
         }
 
+        public static UnmanagedCallingConventions GetDelegateCallingConventions(this TypeDesc delegateType)
+        {
+            Debug.Assert(delegateType.IsDelegate);
+
+            if (delegateType is EcmaType ecmaDelegate)
```

Comment:
Nit: Might want to either do `delegateType.GetTypeDefinition() is EcmaType ecmaDelegate` or assert the type is not generic so that we just don't do bad codegen if we ever allow interop with generic delegates.